pkg/jobs/hive/hive.go

```diff
@@ -181,3 +187,189 @@ func monitorDeployStatus(client clientv1.Client, hiveset hiveclient.Interface, c
 	}
 	return nil
 }
+
+func UpgradeCluster(client clientv1.Client, clusterName string, curator *clustercuratorv1.ClusterCurator) error {
+	klog.V(0).Info("* Initiate Upgrade")
+	klog.V(2).Info("Looking up managedclusterinfo " + clusterName)
+
+	desiredUpdate := curator.Spec.Upgrade.DesiredUpdate
+
+	managedClusterInfo := managedclusterinfov1beta1.ManagedClusterInfo{}
+	if err := client.Get(context.TODO(), types.NamespacedName{
+		Namespace: clusterName,
+		Name:      clusterName,
+	}, &managedClusterInfo); err != nil {
+		return err
+	}
+
+	klog.V(2).Info("kubevendor ", managedClusterInfo.Status.KubeVendor)
+
+	if managedClusterInfo.Status.KubeVendor != "OpenShift" {
+		return errors.New("Can not upgrade non openshift cluster")
+	}
+
+	if desiredUpdate == "" {
+		return errors.New("Provide valid upgrade version")
+	}
+
+	isValidVersion := false
+	if managedClusterInfo.Status.DistributionInfo.OCP.AvailableUpdates != nil {
+		for _, version := range managedClusterInfo.Status.DistributionInfo.OCP.AvailableUpdates {
```

Comment:
so current implementation will not support channel? guess will wait for managedclusterinfo's updates?

Reply:
added channel and upstream. just validation is remaining. will do that in the next PR